tapa/steps/floorplan.py

```diff
@@ -43,40 +43,50 @@ def floorplan(
     fp_slots = []
     if floorplan_path:
         assert program.flattened, "Floorplan can only be applied to a flattened graph"
-        slot_to_insts = get_slot_to_inst(floorplan_path, tapa_graph)
+        slot_to_insts = get_slot_to_inst(floorplan_path, tapa_graph, program)
         tapa_graph = tapa_graph.get_floorplan_graph(slot_to_insts)
         fp_slots = list(slot_to_insts.keys())
 
         graph_dict = tapa_graph.to_dict()
 
-        store_tapa_program(
-            Program(
-                obj=graph_dict,
-                target=program.target,
-                work_dir=program.work_dir,
-                floorplan_slots=fp_slots,
-                flattened=program.flattened,
-            )
+        new_program = Program(
+            obj=graph_dict,
+            target=program.target,
+            work_dir=program.work_dir,
+            floorplan_slots=fp_slots,
+            flattened=program.flattened,
```

Comment:
add a `floorplan_task_name_to_region_mapping=program.floorplan_task_name_to_region_mapping`? Also, is it possible to name it shorter?